src/app/services/quiz.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Quiz } from './../domain/quiz';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class QuizService {
+  quiz: any;
+  questions: any;
+
+  constructor(private httpClient: HttpClient, private route: Router) {}
+
+  // create quiz object
+  createQuiz(initializeForm: any) {
+    this.quiz = new Quiz();
```

Comment:
Services should always be stateless. They can't keep references to an object used multiple times. The service can create a new quiz and return it immediately but can't keep it.